Ninja build failing (clean install)

On Ubuntu 18.04 
Tried with both g++-7.5.0 and then g++-11
After following the exact instructions for building with meson on
[https://github.com/nnstreamer/nntrainer/blob/main/docs/getting-started.md#build-with-meson](https://github.com/nnstreamer/nntrainer/blob/main/docs/getting-started.md#build-with-meson)
ninja -C build install 
Fails with :
[13/408] Compiling C++ object 'nntrain...@nntrainer@sha/ini_interpreter.cpp.o'.
ninja: build stopped: subcommand failed.

```
(base) udit@anon:~/nntrainer$ meson build
The Meson build system
Version: 0.50.1
Source dir: /home/udit/nntrainer
Build dir: /home/udit/nntrainer/build
Build type: native build
Project name: nntrainer
Project version: 0.3.0
Native C compiler: cc (gcc 11.1.0 "cc (Ubuntu 11.1.0-1ubuntu1~18.04.1) 11.1.0")
Native C++ compiler: g++ (gcc 11.1.0 "g++ (Ubuntu 11.1.0-1ubuntu1~18.04.1) 11.1.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Compiler for C supports arguments -Wredundant-decls: YES
Compiler for C++ supports arguments -Wredundant-decls: YES
Compiler for C supports arguments -Wwrite-strings: YES
Compiler for C++ supports arguments -Wwrite-strings: YES
Compiler for C supports arguments -Wformat: YES
Compiler for C++ supports arguments -Wformat: YES
Compiler for C supports arguments -Wformat-nonliteral: YES
Compiler for C++ supports arguments -Wformat-nonliteral: YES
Compiler for C supports arguments -Wformat-security: YES
Compiler for C++ supports arguments -Wformat-security: YES
Compiler for C supports arguments -Winit-self: YES
Compiler for C++ supports arguments -Winit-self: YES
Compiler for C supports arguments -Waddress: YES
Compiler for C++ supports arguments -Waddress: YES
Compiler for C supports arguments -Wno-multichar -Wmultichar: YES
Compiler for C++ supports arguments -Wno-multichar -Wmultichar: YES
Compiler for C supports arguments -Wvla: YES
Compiler for C++ supports arguments -Wvla: YES
Compiler for C supports arguments -Wpointer-arith: YES
Compiler for C++ supports arguments -Wpointer-arith: YES
Compiler for C supports arguments -Wno-error=varargs -Werror=varargs: YES
Compiler for C++ supports arguments -Wno-error=varargs -Werror=varargs: YES
Compiler for C supports arguments -Wdefaulted-function-deleted: NO
Compiler for C++ supports arguments -Wdefaulted-function-deleted: NO
Compiler for C supports arguments -ftree-vectorize: YES
Compiler for C++ supports arguments -ftree-vectorize: YES
Compiler for C supports arguments -Wmissing-declarations: YES
Compiler for C supports arguments -Wmissing-include-dirs: YES
Compiler for C supports arguments -Wmissing-prototypes: YES
Compiler for C supports arguments -Wnested-externs: YES
Compiler for C supports arguments -Waggregate-return: YES
Compiler for C supports arguments -Wold-style-definition: YES
Compiler for C supports arguments -Wdeclaration-after-statement: YES
Compiler for C supports arguments -Wno-error=varargs -Werror=varargs: YES
Program sh found: YES (/bin/sh)
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Dependency blas-openblas found: YES 0.2.20+ds
Dependency OpenMP found: YES 4.5
Library m found: YES
Library dl found: YES
Dependency threads found: YES 
Dependency iniparser found: NO found '3.2' but need: '>=4.1'
Found CMake: /usr/bin/cmake (3.10.2)
Dependency iniparser found: NO (tried pkgconfig and cmake)
Message: falling back to find libiniparser library and header files
Library iniparser found: YES
Has header "iniparser.h" : YES
Dependency capi-ml-inference found: YES 1.8.0
Dependency capi-ml-common found: YES 1.8.0
Dependency tensorflow2-lite found: YES 2.7.0
Dependency GTest found: YES (building self)
Dependency GTest found: YES (building self)
Dependency opencv found: NO (tried pkgconfig and cmake)
Dependency opencv4 found: NO (tried pkgconfig and cmake)
Dependency opencv3 found: NO (tried pkgconfig and cmake)
Program flatc found: YES (/usr/bin/flatc)
Configuring nntrainer.pc using configuration
Configuring nntrainer.ini using configuration
Message: NNTRAINER_CONF_PATH=/usr/local/etc/nntrainer.ini
Message: extra defines are:-DMIN_CPP_VERSION=201703L -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DREDUCE_TOLERANCE=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 -DNNTRAINER_CONF_PATH="/usr/local/etc/nntrainer.ini"
Configuring ccapi-ml-training.pc using configuration
Configuring capi-ml-training.pc using configuration
Dependency jsoncpp found: YES 1.7.4
Dependency libcurl found: YES 7.58.0
Dependency glib-2.0 found: YES 2.56.4
Dependency gstreamer-1.0 found: YES 1.14.5
Dependency nnstreamer found: YES 2.1.0
Dependency glib-2.0 found: YES (cached)
Dependency gmodule-2.0 found: YES 2.56.4
Dependency gstreamer-1.0 found: YES (cached)
Build targets in project: 56
Found ninja-1.8.2 at /usr/bin/ninja
```

```
(base) udit@anon:~/nntrainer$ ninja -C build install
ninja: Entering directory `build'
[4/408] Compiling C++ object 'nntraine...@@nntrainer@sha/data_iteration.cpp.o'.
FAILED: nntrainer/d48ed23@@nntrainer@sha/data_iteration.cpp.o 
g++ -Inntrainer/d48ed23@@nntrainer@sha -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/compiler -I../nntrainer/compiler -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers/loss -I../nntrainer/layers/loss -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I/usr/include/x86_64-linux-gnu -I/usr/include/iniparser -I/usr/include/nnstreamer -I/usr/include/tensorflow2/ -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++17 -O3 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wno-error=varargs -ftree-vectorize -DMIN_CPP_VERSION=201703L -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DREDUCE_TOLERANCE=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 '-DNNTRAINER_CONF_PATH="/usr/local/etc/nntrainer.ini"' -fPIC -fopenmp -pthread -MD -MQ 'nntrainer/d48ed23@@nntrainer@sha/data_iteration.cpp.o' -MF 'nntrainer/d48ed23@@nntrainer@sha/data_iteration.cpp.o.d' -o 'nntrainer/d48ed23@@nntrainer@sha/data_iteration.cpp.o' -c /home/udit/nntrainer/nntrainer/dataset/data_iteration.cpp
In file included from ../nntrainer/dataset/data_iteration.h:21,
                 from /home/udit/nntrainer/nntrainer/dataset/data_iteration.cpp:13:
../nntrainer/tensor/tensor.h: In member function ‘void nntrainer::Tensor::updateBatch(unsigned int)’:
../nntrainer/tensor/tensor.h:1196:18: error: ‘invalid_argument’ is not a member of ‘std’
 1196 |       throw std::invalid_argument(
      |                  ^~~~~~~~~~~~~~~~
[5/408] Compiling C++ object 'nntraine...iner@sha/random_data_producers.cpp.o'.
FAILED: nntrainer/d48ed23@@nntrainer@sha/random_data_producers.cpp.o 
g++ -Inntrainer/d48ed23@@nntrainer@sha -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/compiler -I../nntrainer/compiler -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers/loss -I../nntrainer/layers/loss -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I/usr/include/x86_64-linux-gnu -I/usr/include/iniparser -I/usr/include/nnstreamer -I/usr/include/tensorflow2/ -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++17 -O3 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wno-error=varargs -ftree-vectorize -DMIN_CPP_VERSION=201703L -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DREDUCE_TOLERANCE=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 '-DNNTRAINER_CONF_PATH="/usr/local/etc/nntrainer.ini"' -fPIC -fopenmp -pthread -MD -MQ 'nntrainer/d48ed23@@nntrainer@sha/random_data_producers.cpp.o' -MF 'nntrainer/d48ed23@@nntrainer@sha/random_data_producers.cpp.o.d' -o 'nntrainer/d48ed23@@nntrainer@sha/random_data_producers.cpp.o' -c /home/udit/nntrainer/nntrainer/dataset/random_data_producers.cpp
In file included from ../nntrainer/dataset/data_producer.h:22,
                 from ../nntrainer/dataset/random_data_producers.h:16,
                 from /home/udit/nntrainer/nntrainer/dataset/random_data_producers.cpp:13:
../nntrainer/tensor/tensor.h: In member function ‘void nntrainer::Tensor::updateBatch(unsigned int)’:
../nntrainer/tensor/tensor.h:1196:18: error: ‘invalid_argument’ is not a member of ‘std’
 1196 |       throw std::invalid_argument(
      |                  ^~~~~~~~~~~~~~~~
In file included from ../nntrainer/dataset/random_data_producers.h:16,
                 from /home/udit/nntrainer/nntrainer/dataset/random_data_producers.cpp:13:
../nntrainer/dataset/data_producer.h: In member function ‘virtual void nntrainer::DataProducer::setProperty(const std::vector<std::__cxx11::basic_string<char> >&)’:
../nntrainer/dataset/data_producer.h:73:18: error: ‘invalid_argument’ is not a member of ‘std’
   73 |       throw std::invalid_argument("There are unparsed properties");
      |                  ^~~~~~~~~~~~~~~~
[13/408] Compiling C++ object 'nntrain...@nntrainer@sha/ini_interpreter.cpp.o'.
ninja: build stopped: subcommand failed.

```

Hello! Thanks for the prompt response, 
Here is the error message when I run it with gcc 7.5.0 . 
It's a different C++  error but I thought it was happening because of the C++ version that is why I tested with an upgraded version.
Maybe this error log gives some insight, if it's not building because of some other reason.

```
(base) udit@anon:~/nntrainer$ meson build
The Meson build system
Version: 0.50.1
Source dir: /home/udit/nntrainer
Build dir: /home/udit/nntrainer/build
Build type: native build
Project name: nntrainer
Project version: 0.3.0
Native C compiler: cc (gcc 7.5.0 "cc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0")
Native C++ compiler: c++ (gcc 7.5.0 "c++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Compiler for C supports arguments -Wredundant-decls: YES
Compiler for C++ supports arguments -Wredundant-decls: YES
Compiler for C supports arguments -Wwrite-strings: YES
Compiler for C++ supports arguments -Wwrite-strings: YES
Compiler for C supports arguments -Wformat: YES
Compiler for C++ supports arguments -Wformat: YES
Compiler for C supports arguments -Wformat-nonliteral: YES
Compiler for C++ supports arguments -Wformat-nonliteral: YES
Compiler for C supports arguments -Wformat-security: YES
Compiler for C++ supports arguments -Wformat-security: YES
Compiler for C supports arguments -Winit-self: YES
Compiler for C++ supports arguments -Winit-self: YES
Compiler for C supports arguments -Waddress: YES
Compiler for C++ supports arguments -Waddress: YES
Compiler for C supports arguments -Wno-multichar -Wmultichar: YES
Compiler for C++ supports arguments -Wno-multichar -Wmultichar: YES
Compiler for C supports arguments -Wvla: YES
Compiler for C++ supports arguments -Wvla: YES
Compiler for C supports arguments -Wpointer-arith: YES
Compiler for C++ supports arguments -Wpointer-arith: YES
Compiler for C supports arguments -Wno-error=varargs -Werror=varargs: YES
Compiler for C++ supports arguments -Wno-error=varargs -Werror=varargs: YES
Compiler for C supports arguments -Wdefaulted-function-deleted: NO
Compiler for C++ supports arguments -Wdefaulted-function-deleted: NO
Compiler for C supports arguments -ftree-vectorize: YES
Compiler for C++ supports arguments -ftree-vectorize: YES
Compiler for C supports arguments -Wmissing-declarations: YES
Compiler for C supports arguments -Wmissing-include-dirs: YES
Compiler for C supports arguments -Wmissing-prototypes: YES
Compiler for C supports arguments -Wnested-externs: YES
Compiler for C supports arguments -Waggregate-return: YES
Compiler for C supports arguments -Wold-style-definition: YES
Compiler for C supports arguments -Wdeclaration-after-statement: YES
Compiler for C supports arguments -Wno-error=varargs -Werror=varargs: YES
Program sh found: YES (/bin/sh)
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Dependency blas-openblas found: YES 0.2.20+ds
Dependency OpenMP found: YES 4.5
Library m found: YES
Library dl found: YES
Dependency threads found: YES 
Dependency iniparser found: NO found '3.2' but need: '>=4.1'
Found CMake: /usr/bin/cmake (3.10.2)
Dependency iniparser found: NO (tried pkgconfig and cmake)
Message: falling back to find libiniparser library and header files
Library iniparser found: YES
Has header "iniparser.h" : YES
Dependency capi-ml-inference found: YES 1.8.0
Dependency capi-ml-common found: YES 1.8.0
Dependency tensorflow2-lite found: YES 2.7.0
Dependency GTest found: YES (building self)
Dependency GTest found: YES (building self)
Dependency opencv found: NO (tried pkgconfig and cmake)
Dependency opencv4 found: NO (tried pkgconfig and cmake)
Dependency opencv3 found: NO (tried pkgconfig and cmake)
Program flatc found: YES (/usr/bin/flatc)
Configuring nntrainer.pc using configuration
Configuring nntrainer.ini using configuration
Message: NNTRAINER_CONF_PATH=/usr/local/etc/nntrainer.ini
Message: extra defines are:-DMIN_CPP_VERSION=201703L -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DREDUCE_TOLERANCE=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 -DNNTRAINER_CONF_PATH="/usr/local/etc/nntrainer.ini"
Configuring ccapi-ml-training.pc using configuration
Configuring capi-ml-training.pc using configuration
Dependency jsoncpp found: YES 1.7.4
Dependency libcurl found: YES 7.58.0
Dependency glib-2.0 found: YES 2.56.4
Dependency gstreamer-1.0 found: YES 1.14.5
Dependency nnstreamer found: YES 2.1.0
Dependency glib-2.0 found: YES (cached)
Dependency gmodule-2.0 found: YES 2.56.4
Dependency gstreamer-1.0 found: YES (cached)
Build targets in project: 56
Found ninja-1.8.2 at /usr/bin/ninja

```

```
(base) udit@anon:~/nntrainer$ ninja -C build install
ninja: Entering directory `build'
[9/408] Compiling C++ object 'nntraine...ntrainer@sha/multiout_realizer.cpp.o'.
FAILED: nntrainer/d48ed23@@nntrainer@sha/multiout_realizer.cpp.o 
c++ -Inntrainer/d48ed23@@nntrainer@sha -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/compiler -I../nntrainer/compiler -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers/loss -I../nntrainer/layers/loss -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I/usr/include/x86_64-linux-gnu -I/usr/include/iniparser -I/usr/include/nnstreamer -I/usr/include/tensorflow2/ -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++17 -O3 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wno-error=varargs -ftree-vectorize -DMIN_CPP_VERSION=201703L -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DREDUCE_TOLERANCE=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 '-DNNTRAINER_CONF_PATH="/usr/local/etc/nntrainer.ini"' -fPIC -fopenmp -pthread -MD -MQ 'nntrainer/d48ed23@@nntrainer@sha/multiout_realizer.cpp.o' -MF 'nntrainer/d48ed23@@nntrainer@sha/multiout_realizer.cpp.o.d' -o 'nntrainer/d48ed23@@nntrainer@sha/multiout_realizer.cpp.o' -c /home/udit/nntrainer/nntrainer/compiler/multiout_realizer.cpp
/home/udit/nntrainer/nntrainer/compiler/multiout_realizer.cpp: In member function ‘virtual nntrainer::GraphRepresentation nntrainer::MultioutRealizer::realize(const GraphRepresentation&)’:
/home/udit/nntrainer/nntrainer/compiler/multiout_realizer.cpp:45:42: error: unused variable ‘result’ [-Werror=unused-variable]
       [[maybe_unused]] auto [iter, result] = freq_map.try_emplace(c, 0);
                                          ^
cc1plus: all warnings being treated as errors
[13/408] Compiling C++ object 'nntrain...trainer@sha/recurrent_realizer.cpp.o'.
FAILED: nntrainer/d48ed23@@nntrainer@sha/recurrent_realizer.cpp.o 
c++ -Inntrainer/d48ed23@@nntrainer@sha -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/compiler -I../nntrainer/compiler -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers/loss -I../nntrainer/layers/loss -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I/usr/include/x86_64-linux-gnu -I/usr/include/iniparser -I/usr/include/nnstreamer -I/usr/include/tensorflow2/ -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++17 -O3 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wno-error=varargs -ftree-vectorize -DMIN_CPP_VERSION=201703L -DUSE_BLAS=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DREDUCE_TOLERANCE=1 -DNNSTREAMER_AVAILABLE=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 '-DNNTRAINER_CONF_PATH="/usr/local/etc/nntrainer.ini"' -fPIC -fopenmp -pthread -MD -MQ 'nntrainer/d48ed23@@nntrainer@sha/recurrent_realizer.cpp.o' -MF 'nntrainer/d48ed23@@nntrainer@sha/recurrent_realizer.cpp.o.d' -o 'nntrainer/d48ed23@@nntrainer@sha/recurrent_realizer.cpp.o' -c /home/udit/nntrainer/nntrainer/compiler/recurrent_realizer.cpp
/home/udit/nntrainer/nntrainer/compiler/recurrent_realizer.cpp: In constructor ‘nntrainer::RecurrentRealizer::RecurrentRealizer(const std::vector<std::__cxx11::basic_string<char> >&, const std::vector<nntrainer::Connection>&, const std::vector<nntrainer::Connection>&)’:
/home/udit/nntrainer/nntrainer/compiler/recurrent_realizer.cpp:134:64: error: unused variable ‘unroll_for’ [-Werror=unused-variable]
   auto &[inputs, outputs, as_sequence, unroll_for, input_is_seq] =
                                                                ^
cc1plus: all warnings being treated as errors
[18/408] Compiling C++ object 'nntrain...trainer@sha/tflite_interpreter.cpp.o'.
ninja: build stopped: subcommand failed.
```

--- a/meson.build
+++ b/meson.build
@@ -39,7 +39,8 @@ warning_flags = [
   '-Wpointer-arith',
   '-Wno-error=varargs',
   '-Wdefaulted-function-deleted',
-  '-ftree-vectorize'
+  '-ftree-vectorize',
+  '-Wno-unused-variable'
 ]
 
 warning_c_flags = [

--- a/nntrainer/tensor/memory_pool.cpp
+++ b/nntrainer/tensor/memory_pool.cpp
@@ -12,6 +12,7 @@
 
 #include <numeric>
 #include <vector>
+#include <limits>
 
 #include <memory_pool.h>
 #include <nntrainer_error.h>

--- a/nntrainer/tensor/tensor.h
+++ b/nntrainer/tensor/tensor.h
@@ -30,6 +30,7 @@
 #include <functional>
 #include <memory>
 #include <vector>
+#include <stdexcept>
 
 #include <tensor_dim.h>
 

--- a/test/unittest/layers/meson.build
+++ b/test/unittest/layers/meson.build
@@ -5,6 +5,7 @@ layer_common_test_dependent_files = files('layers_dependent_common_tests.cpp')
 nntrainer_layer_common_standalone_tests_lib = shared_library(
   'nntrainer_layer_common_standalone_tests',
   'layers_standalone_common_tests.cpp',
+  cpp_args : '-Wno-maybe-uninitialized',
   dependencies: [nntrainer_dep, gtest_dep], # nntrainer_devel_dep
   include_directories: layer_common_test_inc
 )
@@ -17,6 +18,7 @@ nntrainer_layer_common_standalone_tests_dep = declare_dependency(
 nntrainer_layer_common_dependent_tests_lib = shared_library(
   'nntrainer_layer_common_dependent_tests',
   'layers_dependent_common_tests.cpp',
+  cpp_args : '-Wno-maybe-uninitialized',
   dependencies: [nntrainer_dep, gtest_dep], # nntrainer_devel_dep
   include_directories: layer_common_test_inc
 )
@@ -80,6 +82,7 @@ exe = executable(
   dependencies: [
     nntrainer_test_main_deps,
   ],
+  cpp_args : '-Wno-maybe-uninitialized',
   install: get_option('enable-test'),
   install_dir: application_install_dir
 )

--- a/test/unittest/memory/meson.build
+++ b/test/unittest/memory/meson.build
@@ -3,6 +3,7 @@ memory_test_inc = include_directories('./')
 nntrainer_memory_planner_tests_lib = static_library(
   'nntrainer_memory_planner_validation',
   'memory_planner_validate.cpp',
+  cpp_args : '-Wno-maybe-uninitialized',
   dependencies: [nntrainer_dep, gtest_dep], # nntrainer_devel_dep
   include_directories: memory_test_inc
 )
@@ -24,6 +25,7 @@ exe = executable(
   dependencies: [
     nntrainer_test_main_deps
   ],
+  cpp_args : '-Wno-maybe-uninitialized',
   install: get_option('enable-test'),
   install_dir: application_install_dir
 )

--- a/test/unittest/memory/unittest_memory_pool.cpp
+++ b/test/unittest/memory/unittest_memory_pool.cpp
@@ -402,7 +402,7 @@ TEST(MemoryPool, get_memory_03_n) {
  */
 TEST(MemoryPool, get_memory_04_p) {
   nntrainer::MemoryPool pool;
-  void *mem;
+  void *mem = nullptr;
 
   auto idx = pool.requestMemory(1, 4, 5);
   EXPECT_NO_THROW(pool.planLayout(nntrainer::BasicPlanner()));

